src/Agent/NewRelic/Agent/Core/Configuration/AppSettingsConfigResolveWhenUsed.cs

```diff
@@ -24,24 +25,33 @@ private static IConfiguration Configuration
         private static IConfigurationRoot InitializeConfiguration()
         {
             var env = new SystemInterfaces.Environment();
-            var currentDirectory = Directory.GetCurrentDirectory();
+            var applicationDirectory = string.Empty;
+            try
+            {
+                applicationDirectory = AppDomain.CurrentDomain.BaseDirectory;
+            }
+            catch (AppDomainUnloadedException)
+            {
+                // Fall back to previous behavior
+                applicationDirectory = Directory.GetCurrentDirectory();
+            }
             var environment = env.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
```

Comment:
Code org nit: Can we move `var env` declaration to right before where it is used on line 38?
Similarly, Can we move the block that sets/checks `var environment` to right before it is used on line 48?